src/blocks/recommendations/components/parsely-recommendations-fetcher.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * External dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { useDebounce } from '@wordpress/compose';
+import { useCallback, useEffect } from '@wordpress/element';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import { setError, setRecommendations } from '../actions';
+import { useRecommendationsStore } from '../recommendations-store';
+
+const ParselyRecommendationsFetcher = ( { boost, limit, sort } ) => {
+	const {
+		state: {},
+		dispatch,
+	} = useRecommendationsStore();
+
+	const query = {
+		boost,
+		limit,
+		sort,
+		url: window.location.href,
+	};
+
+	async function fetchRecosFromWpApi() {
+		return apiFetch( {
+			path: addQueryArgs( '/wp-parsely/v1/related', { query } ),
+		} );
+	}
+
+	async function fetchRecos() {
+		let response;
+		let errorMessage;
+
+		try {
+			response = await fetchRecosFromWpApi();
+		} catch ( wpError ) {
+			errorMessage = wpError;
+		}
+
+		if ( response?.error ) {
+			errorMessage = response.error;
+		}
+
+		if ( errorMessage ) {
+			dispatch( setError( { error: response.error } ) );
+			return;
+		}
+
+		const data = response?.data || [];
+		dispatch( setRecommendations( { recommendations: data } ) );
+	}
+
+	const apiMemoProps = [ ...Object.values( query ) ];
+
+	const updateRecosWhenPropsChange = useCallback( fetchRecos, apiMemoProps );
+
+	const debouncedUpdate = useDebounce( updateRecosWhenPropsChange, 300 );
```

Comment:
Can this magic number be extracted into a variable, or be commented to say what it means? (Presumably 300 seconds?)

Reply:
This seems to be the minimum milliseconds of wait before re-invoking the function (if needed), to ensure that calls are not executed too frequently. So in this case, even if props were to change every 50ms, we would be calling the respective function every 300ms.

I've updated the code but left this open for visibility, feel free to comment on this or mark it as resolved. As a bonus, I've changed "Recos" to "Recommendations".